pkg/validator/validator.go

```diff
@@ -0,0 +1,131 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package validator
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+)
+
+type Validator struct {
+	// Baseurl of the jira API
+	// Example https://your-domain.atlassian.net/rest/api/3
+	baseURL string
+}
+
+type JiraIssue struct {
+	Key string `json:"key"`
+	ID  string `json:"id"`
+}
+
+type Match struct {
+	MatchedIssues []int    `json:"matchedIssues"`
+	Errors        []string `json:"errors"`
+}
+
+type MatchResult struct {
+	Matches []Match `json:"matches"`
+}
+
+func (v *Validator) MatchIssue(ctx context.Context, issueKey, jql, jiraAccount, apiToken string) (*MatchResult, error) {
```

Comment:
API token (probably also jiraAccount) will be stored in secret manager and the caller will request it from secret manager on each rpc. It is possible that the value in secret manager will change between requests, so I am making it an input param instead of a field in the validator. Let me what you think.

Reply:
Hey @dandans-verily sorry, I'm not sure I understand. Do we expect the JIRA API key to change frequently? Is each employee going to use their own identity to access the JIRA API key?

A more common architecture would be:

- JIRA plugin runs as a dedicated IAM service account identity
- Identity has permissions to access Secret Manager secrets
- At boot (or via a native integration), the service accesses all the secrets from Secret Manager and stores them in memory
- Either secrets are periodically refreshed or the secrets are cached indefinitely until the next restart

Reply:
The API token rotation cadence is not decided yet, so I was thinking to start with "no cache" in the first version. Your comment makes sense to me, we can definitely start with "cached indefinitely until the next restart" for now. 

I also realized jql is with similar nature, its value will be decided at deployment time, so change it to a field as well.